app/refund/refund.go

```diff
@@ -23,10 +24,18 @@ func NewGasRefundHandler(ak auth.AccountKeeper, sk types.SupplyKeeper, ik innert
 	) (refundFee sdk.Coins, err error) {
 		var gasRefundHandler sdk.GasRefundHandler
 
-		if tx.GetType() == sdk.EvmTxType {
-			gasRefundHandler = evmGasRefundHandler
+		if tmtypes.HigherThanEarth(ctx.BlockHeight()) {
+			if tx.GetType() == sdk.EvmTxType || tx.GetType() == sdk.StdTxType {
+				gasRefundHandler = evmGasRefundHandler
+			} else {
+				return nil, nil
+			}
```

Comment:
```suggestion
		if tmtypes.HigherThanEarth(ctx.BlockHeight()) {
				gasRefundHandler = evmGasRefundHandler
```

Reply:
1. no need to check tx type, because type must be EvmTxType or StdTxType, otherwise tx won`t be submitted.
2. reanme evmGasRefundHandler to gasRefundHandler